arm-software/linux/build.sh-HOWTO.md

```diff
@@ -0,0 +1,163 @@
+# The Arm Toolchain for Linux building script (build.sh)
+
+The aim of the `build.sh` script is to build the ATfL product and package it
+into a tarball. Usually it is being started in a CI environment which is
+responsible for providing all of the necessary building tools and setting
+the environment variables it utilizes. It should be also possible to run
+this script manually and this document should explain how to do that on
+various Linux distribution.
+
+## Preparation of the building environment
+
+### `.deb`-based systems (Debian, Ubuntu)
+
+The following packages need to be installed (using `apt`):
+
+- `binutils-dev`
+- `build-essential`
+- `cmake`
+- `figlet`
+- `git`
+- `libzstd-dev`
+- `ninja-build`
+- `python3`
+- `python3-dev`
+- `python3-myst-parser`
+- `python3-pip`
+- `python3-pygments`
+- `python3-yaml`
+- `zlib1g-dev`
+
+### `.rpm`-based systems (RHEL, CentOS, Alma, Amazon)
+
+- `libgcc`
+- `gcc`
+- `gcc-c++`
+- `diffutils`
+- `patch`
+- `binutils`
+- `binutils-devel`
+- `make`
+- `cmake`
+- `figlet`
+- `git`
+- `ninja-build`
+- `zlib-devel`
+- `zlib-static`
+- `zstd`
+- `wget`
+- `mpfr`
+- `mpfr-devel`
+- `libmpc`
+- `libmpc-devel`
+- `isl`
+- `isl-devel`
+- `cpio`
+
+#### Additionally, on older `.rpm`-based systems (RHEL8, CentOS8):
+
+- `python38`
+- `python38-devel`
+- `python38-pip`
+- `python38-pyyaml`
+
+#### Additionally, on more recent `.rpm`-based systems:
+
+- `python3`
+- `python3-devel`
+- `python3-pip`
+- `python3-pygments`
+- `python3-pyyaml`
+- `python3-myst-parser` (since RHEL10)
+
+## Optional: obtaining libamath.a and libamath.so
+
+The Amath library is used by the loop vectorizer to vectorize the loops from
+which math functions are being called. This library is distributed along with
+Arm Performance Libraries.
+
+On the `.deb`-based systems, the easiest way to obtain and extract `libamath.a`
+and `libamath.so` (current latest version, 24.10) is to do the following:
+
+```
+$ mkdir libamath
+
+$ cd libamath
+
+$ wget https://developer.arm.com/-/cdn-downloads/permalink/Arm-Performance-Libraries/Version_24.10/arm-performance-libraries_24.10_deb_flang-new.tar
+
+$ tar -xf arm-performance-libraries_24.10_deb_flang-new.tar
+
+$ arm-performance-libraries_24.10_deb/arm-performance-libraries_24.10_deb.sh -a -f -s .
```

Comment:
nit: are you sure it is okay to recommend people automatically accept the EULA (`-a`)? Perhaps there should be a note in this HOWTO just in case.

Reply:
Discussed offline.